modules/caddyhttp/reverseproxy/hosts.go

```diff
@@ -259,3 +259,15 @@ var hosts = caddy.NewUsagePool()
 // dialInfoVarKey is the key used for the variable that holds
 // the dial info for the upstream connection.
 const dialInfoVarKey = "reverse_proxy.dial_info"
+
+// proxyProtocolInfoVarKey is the key used for the variable that holds
+// the proxy protocol info for the upstream connection.
+const proxyProtocolInfoVarKey = "reverse_proxy.proxy_protocol_info"
+
+// ProxyProtocolInfo contains information needed to write proxy protocol to a
+// connection to an upstream host.
+type ProxyProtocolInfo struct {
```

Comment:
We should change the type to [netip.AddrPort](https://pkg.go.dev/net/netip#AddrPort)

Reply:
@WeidiDeng If we do so we'll have to convert it back at https://github.com/caddyserver/caddy/pull/5424/files#diff-8839c76ab9a0e424d77cabc5357fbcdfb166d6b6cc7a076114fb24a5b16814b8R223 anyway because the proxyprotocol.Header needs the current types. Also, the conversion would have to be done in some hacky way, imo (didn't see any native method to convert between the types). So is it really worth the effort?

Reply:
[AsSlice](https://pkg.go.dev/net/netip#Addr.AsSlice) and [Port](https://pkg.go.dev/net/netip#AddrPort.Port) are what you need.

Reply:
@WeidiDeng Thanks, done now.